apps/passport-client/components/screens/FrogScreens/GetFrogTab.tsx

```diff
@@ -0,0 +1,176 @@
+        if (previous && previous.id !== latest.id) {
+          setSelectedPCDID(latest.id);
+        }
+
+        return latest;
+      });
+    }
+  }, [frogPCDs]);
+
+  const [searchMessage, setSearchMessage] = useState("");
+  useEffect(() => {
+    if (searchMessage) {
+      const timeout = setTimeout(() => {
+        setSearchMessage("");
+      }, 5000);
+
+      return () => {
+        clearTimeout(timeout);
+      };
+    }
+  }, [searchMessage]);
+
+  return (
+    <>
+      <SearchGroup>
+        {subs.map((sub) => (
+          <SearchButton
+            key={sub.id}
+            sub={sub}
+            refetch={refetch}
+            setMessage={setSearchMessage}
+          />
+        ))}
+      </SearchGroup>
+
+      {searchMessage && <Notice>{searchMessage}</Notice>}
+
+      {frogPCDs.length > 0 && (
+        <>
+          <Separator style={{ margin: 0 }} />
+          <PCDContainer>
+            {frogPCDs.map((pcd) => (
+              <PCDCard
+                key={pcd.id}
+                pcd={pcd}
+                onClick={onPcdClick}
+                expanded={pcd.id === selectedPCD?.id}
+              />
+            ))}
+          </PCDContainer>
+        </>
+      )}
+    </>
+  );
+}
+
+const SearchButton = ({
+  sub: { id, feed, nextFetchAt },
+  refetch,
+  setMessage
+}: {
+  sub: Subscription & { nextFetchAt?: number };
+  refetch: () => Promise<void>;
+  setMessage: (message: string) => void;
+}) => {
+  const dispatch = useDispatch();
+  const countDown = useCountDown(nextFetchAt || 0);
+  const canFetch = !nextFetchAt || nextFetchAt < Date.now();
+
+  const onClick = useCallback(
+    () =>
+      new Promise<void>((resolve, reject) => {
+        dispatch({
+          type: "sync-subscription",
+          subscriptionId: id,
+          onSucess: () => {
+            setMessage(`You found a new frog in ${feed.name}!`);
+            refetch().then(resolve).catch(reject);
+          },
+          onError: (e) => refetch().finally(() => reject(e))
+        });
+      }),
+    [dispatch, feed.name, id, refetch, setMessage]
+  );
+  const name = useMemo(() => _.upperCase(`Search ${feed.name}`), [feed.name]);
+
+  return (
+    <ActionButton key={id} onClick={onClick} disabled={!canFetch}>
+      {canFetch ? name : `${name}${countDown}`}
+    </ActionButton>
+  );
+};
+
+function useCountDown(timestamp: number) {
+  const end = useMemo(() => new Date(timestamp), [timestamp]);
+  const [diffText, setDiffText] = useState("");
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      const now = new Date();
+      const diffMs = Math.ceil((end.getTime() - now.getTime()) / 1000) * 1000;
+      if (diffMs <= 0) {
+        setDiffText("");
+      } else {
+        const diffString = prettyMilliseconds(diffMs, {
+          millisecondsDecimalDigits: 0,
+          secondsDecimalDigits: 0,
+          unitCount: 4
+        });
+        setDiffText(diffString);
+      }
+    }, 1000);
```

Comment:
suggest 

```suggestion
    }, 500);
```

otherwise there's some chance it skips a second or freezes for a second